src/relay/pass/inline.cc

```diff
@@ -0,0 +1,227 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file tvm/relay/pass/inline.cc
+ * \brief Global function inliner. It contains the following steps:
+ *
+ *  - Preprocessing: eligibility checking. Only inline the functions that can
+ *  be inlined. We currently only use simple rules to make the decision. No
+ *  profitibility analysis is available for now.
+ *
+ *  - Inline: replace the call with a function or the function body depending on
```

Comment:
I don't understand what inline does to external functions.  I was expecting that Inline should be nop for external functions. What does `replace the call with a function` here mean in this case?

Reply:
For example:

‘’’python 
Let x = @foo 
‘’’
Becomes
‘’’python 
Let x = function(...) { body } # function doesn’t use default compiler 
‘’’

Or 
‘’’python 
Let x = body # for normal functions we replace call e function body, need to take care of args 
‘’’

Reply:
Do you mean it is replaced with an anonymous function that immediately gets executed?

Reply:
Yes, this could help composite as well